scripts/ci/generate_install_lists.py

```diff
@@ -119,9 +121,12 @@ def main(argv=sys.argv[1:]):
         if args.skip_rosdep_keys:
             dependency_keys_build.difference_update(args.skip_rosdep_keys)
             dependency_keys_test.difference_update(args.skip_rosdep_keys)
-        if args.package_selection_args:
-            dependency_keys_build.difference_update(to_ignore)
-            dependency_keys_test.difference_update(to_ignore)
+
+        # remove all non-ROS packages and packages which are present but
+        # specifically ignored
+        every_package_name = all_packages.keys() | all_underlay_pkg_names
```

Comment:
This is the actual bugfix.

During the dependency walk, any ROS packages encountered as a dependency were recursively walked, and therefore aren't included in the resulting enumeration. Beyond that, we remove any keys which have been specifically ignored (via `skip_rosdep_keys`). Dirk then added another quirk that if a package is present but was specifically ignored, we also exclude it from the dependencies to avoid installing a deb package to satisfy the dependency.

The change is to take that a step further and remove dependencies on any non-ROS packages present in the workspaces. Instead of specifically enumerating the non-ROS packages, I'm just removing any package (ROS, non-ROS, ignored, etc) present in the workspace.

The new behavior should be that the only time a ROS deb is installed is if the package isn't part of ANY of the workspaces AND the package isn't specifically ignored via `skip_rosdep_keys`.

Reply:
Thanks for elaborating. To me this kind of info is critical to go in PR discussions or ideally IMO commit messages so it's there for future editors to review.

Reply:
Thanks for the feedback. I'll drop this in the squashed commit message :)